### PR TITLE
Kafka Connect Avro Support Part 3: Basic Avro support

### PR DESCRIPTION
**Pull request by liuzix**
### What problem does this PR solve?
#660 

### What is changed and how it works?
MqSink now supports protocol "avro".

### Check List 

Tests 

 - Unit test
 - Manual test: Syncing through Kafka JDBC sink connector tested.

Code changes
 - Has interface methods change

Related changes

    - Need to update the documentation
    - Replicating date or timestamp data may not work for now.
    - Enable it by adding "protocol=avro" to sink-uri and "-opts registry=http://..." when creating change feed.

### Release note

- Add Avro support in mq sink, which enables Kafka sink connector to work together with TiCDC.

### CODE REVIEW

File: cdc/sink/mq.go [hunk 1]
```diff
@@ -163,7 +184,7 @@ flushLoop:
 
 func (k *mqSink) EmitCheckpointTs(ctx context.Context, ts uint64) error {
 	switch k.protocol {
-	case codec.ProtocolCanal: // ignore resolved events in canal protocol
+	case codec.ProtocolCanal, codec.ProtocolAvro: // ignore resolved events in canal protocol
```
Comment (amyangfei) [inline cdc/sink/mq.go:187]:
```suggestion
	case codec.ProtocolCanal, codec.ProtocolAvro: // ignore resolved events in canal and avro protocol
```

File: cdc/sink/mq.go [hunk 2]
```diff
@@ -194,11 +215,60 @@ func (k *mqSink) EmitDDLEvent(ctx context.Context, ddl *model.DDLEvent) error {
...
+			return errors.New("No schema manager in Avro encoder, probably bug")
+		}
+
+		for _, tableName := range schemaSnap.CloneTables() {
+			info, ok := schemaSnap.GetTableByName(tableName.Schema, tableName.Table)
+			if !ok {
+				return errors.Errorf("Table %s not found, probably bug", tableName.String())
+			}
+
+			if !info.IsEligible() || tableName.Schema == "mysql" {
```
Comment (amyangfei) [inline cdc/sink/mq.go:245]:
```suggestion
			if !info.IsEligible() || k.filter.ShouldIgnoreTable(tableName.Schema, tableName.Table) {
```

File: cdc/sink/mq.go [hunk 3]
```diff
@@ -194,11 +215,60 @@ func (k *mqSink) EmitDDLEvent(ctx context.Context, ddl *model.DDLEvent) error {
...
+
+			columnInfo := make([]*model.ColumnInfo, len(info.Cols()))
+			for i := range columnInfo {
+				columnInfo[i] = new(model.ColumnInfo)
+				columnInfo[i].FromTiColumnInfo(info.Cols()[i])
+			}
+
+			str, err := codec.ColumnInfoToAvroSchema(tableName.Table, columnInfo)
+			if err != nil {
+				return errors.New("Error in Initialize")
```
Comment (amyangfei) [inline cdc/sink/mq.go:258]:
```suggestion
				return errors. Annotate(err, "Error in Initialize")
```

**Comment by amyangfei** [comment]
Please merge master and resolve conflict

File: cdc/model/sink.go [hunk 4]
```diff
@@ -102,29 +102,51 @@ func (c *ColumnInfo) FromTiColumnInfo(tiColumnInfo *model.ColumnInfo) {
...
-	CommitTs   uint64
-	Schema     string
-	SchemaID   int64
-	Table      string
-	ColumnInfo []*ColumnInfo
-	Query      string
-	Type       model.ActionType
+	StartTs   uint64
+	CommitTs  uint64
+	Schema    string
```
Comment (leoppro) [inline cdc/model/sink.go:120]: we can remove L119-L120 now

**Comment by leoppro** [comment]
please fix build

File: cdc/model/sink.go [hunk 5]
```diff
@@ -100,30 +102,52 @@ func (c *ColumnInfo) FromTiColumnInfo(tiColumnInfo *model.ColumnInfo) {
...
+			e.TableInfo.ColumnInfo[i] = new(ColumnInfo)
+			e.TableInfo.ColumnInfo[i].FromTiColumnInfo(colInfo)
+		}
+
+		e.TableInfo.Schema = job.SchemaName
+		e.TableInfo.Table = tableName
+		e.TableInfo.UpdateTs = tableInfo.UpdateTS
+		e.Table = tableName
+	} else {
+		log.Warn("BinlogInfo.TableInfo is null")
```
Comment (leoppro) [inline cdc/model/sink.go:145]:
```suggestion
```

**Pull request merged**